### PR TITLE
Enhance error message for copy-to

### LINKED ISSUE #49344: copy_to breaks on date_range type
**Elasticsearch version** (`bin/elasticsearch --version`): 7.4.2

**Description of the problem including expected versus actual behavior**:

Using `copy_to` with a `date_range` field causes errors on ingest. Interestingly (worse?), if you define a field that is not mapped and mappings are disabled, then it silently ignores the `copy_to` request.

**Steps to reproduce**:

 1\. Create the index

```http
PUT /test
{
  "mappings": {
    "properties": {
      "date_copy": {
        "type": "date_range"
      },
      "date": {
        "type": "date_range",
        "copy_to": "date_copy"
      }
    }
  }
}
```

 2\. Attempt to index a document.
 
```http
PUT /test/_doc/1
{
  "date": {
    "gte": "2019-11-10T01:00:00.000Z",
    "lt": "2020-01-01"
  }
}
```

3\. Observe error.

```json
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "error parsing field [date_copy], expected an object but got date"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "failed to parse field [date_copy] of type [date_range] in document with id '1'. Preview of field's value: 'null'",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "error parsing field [date_copy], expected an object but got date"
    }
  },
  "status": 400
}
```

### PR DESCRIPTION
**Pull request by cbuescher**
We currently don't support `copy_to` for fields that take the form of objects
(e.g. `date_range` or certain kinds of `geo_point` variants). The current
problem with objects is that when DocumentParser parses anything other than
single values, it potentially advances the underlying parser past the value that
we would need to stay on for parsing the value again. While we might want to
support this in the future, for now this PR enhances the otherwise confusing
MapperParsingException with something more helpful and adds a short note in the
documentation about this restriction.

Closes #49344

### CODE REVIEW

**Review by jtibshirani — APPROVED** [review]
This looks good to me. It's great the check is non-intrusive.

I left one suggestion (not critical) for making the docs note more visible.

**Comment by cbuescher** [comment]
@elasticmachine update branch

File: docs/reference/mapping/params/copy-to.asciidoc [hunk 1]
```diff
@@ -67,4 +67,5 @@ Some important points:
 * You cannot copy recursively via intermediary fields such as a `copy_to` on 
 `field_1` to `field_2` and `copy_to` on `field_2` to `field_3` expecting 
 indexing into `field_1` will eventuate in `field_3`, instead use copy_to 
-directly to multiple fields from the originating field. 
\ No newline at end of file
+directly to multiple fields from the originating field.
```
[the remark this answers is not in the document]
Comment (cbuescher) [reply docs/reference/mapping/params/copy-to.asciidoc:70]: Thanks for the suggestion, I changed that.

**Comment by freost** [comment]
This seems to trigger an error when copying `null` values. Was this intentional?

**Comment by jtibshirani** [comment]
@freost thank you for catching this, it was definitely not intentional. I've opened #76665 to fix this in the next patch release.

**Pull request merged**